src/Controllers/ImageController.php

```diff
@@ -72,6 +72,65 @@ public function __invoke(Request $request, int $store, string $size, string $pla
         return $this->storage()->response($resizedPath);
     }
 
+    public function getResizedPath(string $size, string $file, string $webp)
+    {
+        foreach (config('imageresizer.external') as $placeholder => $url) {
+            if (Str::startsWith($file, $placeholder)) {
+                $file = Str::replaceFirst($placeholder, '', $file);
+                $placeholderUrl = $url;
+                break;
+            }
+        }
+
+        $placeholder = isset($placeholderUrl)
+            ? $placeholder
+            : 'local';
+
+        return config('rapidez.store').'/resizes/'.$placeholder.'/'.$size.'/'.$file.$webp;
+    }
+
+    public function redirectFromSku(Request $request, string $size, string $file)
+    {
+        $webp = str_ends_with($file, '.webp') ? '.webp' : '';
+        $baseFile = $webp ? str_replace_last('.webp', '', $file) : $file;
+
+        $sku = pathinfo($baseFile)['filename'];
+        $image = $this->productImageUrlFromSku($sku);
+
+        if ($image == 'magento/catalog/placeholder.jpg') {
+            return redirect($this->getResizedPath($size, $image, $webp));
+        }
+
+        $pathSku = $this->getResizedPath($size, 'magento/sku/'.$baseFile, $webp);
+        $pathImage = $this->getResizedPath($size, $image, $webp);
+
+        if (!$this->storage()->directoryExists(dirname($pathSku))) {
+            $this->storage()->createDirectory(dirname($pathSku));
+        }
```

Comment:
I thought Flysystem handles the creation of un-existing directories automatically?